src/test/java/mate/academy/security/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,95 @@
+package mate.academy.security;
+
+import static mate.academy.model.Role.RoleName.USER;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.RoleService;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class AuthenticationServiceTest {
+    private UserService userService;
+    private RoleService roleService;
+    private PasswordEncoder passwordEncoder;
+    private AuthenticationService authenticationService;
+    private String email = "user@gmail.com";
+    private String password = "password";
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        userService = Mockito.mock(UserService.class);
+        roleService = Mockito.mock(RoleService.class);
+        passwordEncoder = Mockito.mock(PasswordEncoder.class);
+        authenticationService =
+                new AuthenticationServiceImpl(userService, roleService, passwordEncoder);
+        user = new User();
+        user.setEmail(email);
+        user.setPassword(password);
+        user.setRoles(Set.of(new Role(USER)));
+    }
+
+    @Test
+    void register_ValidData_Ok() {
+        Mockito.when(roleService.getRoleByName("USER")).thenReturn(new Role(USER));
+        Mockito.when(userService.save(user)).thenReturn(user);
+        User actual = authenticationService.register(email, password);
+        assertNotNull(actual);
+        assertEquals(user, actual);
+    }
+
+    @Test
+    void register_EmailIsNull_NotOk() {
+        try {
+            authenticationService.register(null, password);
+        } catch (RuntimeException e) {
+            assertEquals("Email and password can't be null", e.getMessage());
+        }

```

Comment:
Please rework all try/catches with assertThrows methods:
```suggestion
        AuthenticationException ex = assertThrows(AuthenticationException.class,
                () -> authenticationService.register(null, password));
        assertEquals("Email and password can't be null", ex.getMessage());
```